src/probe/probe.c

```diff
@@ -1491,6 +1491,70 @@ static int probe_set_config(struct processing_module *mod, uint32_t param_id,
 	}
 }
 
+static int probe_add_point_info_params(struct sof_ipc_probe_info_params *info,
+				       probe_point_id_t id, int index, size_t max_size)
+{
+	struct probe_pdata *_probe = probe_get();
+	struct probe_point pp = {
+		.buffer_id = id,
+		.purpose = PROBE_PURPOSE_EXTRACTION,
+	};
+	int i;
+
+	if (offsetof(struct sof_ipc_probe_info_params, probe_point[index]) +
+	    sizeof(pp) > max_size) {
+		info->num_elems = index;
+		return -1;
```

Comment:
proper error code? I see that it isn't propagated (for now) but (1) it might get propagated in the future, and (2) -1 just doesn't make much sense - it looks like a magic constant. Something like `-ENOMEM` or `-ENOSPC` maybe